### PR TITLE
[M1_TR-3] Setup CI/CD and checkers

### PR DESCRIPTION
**Pull request by jasonchuasun**
### Related Issue
- [Backlog](https://framgiaph.backlog.com/view/M1_TR-3)

### Context
- set up .sun-ci.yml file for running build, lint, and test. 
- When creating a PR, this will check for any failures first before allowing merging.

#### commands: 
  - install devDependency - husky. In root dir, run the command `npm i`

### Test Executed
- [ ] Unit Test
- [ ] Functional Test

### Screenshot
Screenshot 2023-08-25 at 10 23 17 AM
Screenshot 2023-08-25 at 10 23 33 AM

### Nature of task
- [ ] Technical Setup

### Risks
- [ ] Affecting other functions

### Notes
- the `functionFile.spec.ts` file was updated to contain a test that always passes (for CI/CD demo purposes only)

### CODE REVIEW

File: package-lock.json
Comment (jeremiahC) [inline package-lock.json:1]: we will be using yarn.lock more, so please delete

File: web/package-lock.json
Comment (jeremiahC) [inline web/package-lock.json:1]: Please delete since we will use yarn

**Review by jeremiahC — CHANGES_REQUESTED** [review]
Please add test_result and coverage. Like this:
```
  stage: test
  image: sunasteriskrnd/php-workspace:7.4
  workspace: shared
  script:
  - phpunit --coverage-clover ./coverage.xml --log-junit ./junit.xml
  coverage:
    type: clover
    path: coverage.xml
  test_result:
    type: junit
    path: junit.xml  
  depends_on:
    - build php
```

Reference: https://ci.sun-asterisk.com/docs/guide/yaml/#configuration-example

**Review by arden-dave — DISMISSED** [review]

**Review by nicoleamber — DISMISSED** [review]

**Review by jeremiahC — APPROVED** [review]

**Review by NiloJr-sun — DISMISSED** [review]

**Review by jeremiahC — CHANGES_REQUESTED** [review]
- I forgot to comment, please add `post-lint` stage. Even though we have husky for pre-commit. Just another layer of checking, please add linter in the ci config

File: .sun-ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    image: node:16-alpine
+    cache:
+      - key:
+          files:
+            - server/yarn.lock
+        paths:
+          - server/node_modules
+    script:
+      - cd server
+      - yarn install
```
Comment (jeremiahC) [inline .sun-ci.yml:24]: setting up the environment should be done on the build stage. Also, Please include frontend as well, you can separete the stages for testing FE and BE.

File: .sun-ci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  - build
+  - test
+
+jobs: 
+  build_and_lint:
+    stage: build
+    image: node:16-alpine
+    script: 
+      - yarn install --frozen-lockfile
+      - yarn eslint .
```
Comment (jeremiahC) [inline .sun-ci.yml:11]: can you separate the stages

- build (preparing your backend and fe)
   - Suggestion: if you want to run one command for all, you can create a bash sript and will run both fe and be commands
- post-lint (for running linter)
- test-be 
- test-fe

File: .sun-ci.yml
Comment (jeremiahC) [inline .sun-ci.yml:1]: For the yarn install, can it be done on the build stage? I think you don't need to yarn install all the time in different stages.

**Review by jeremiahC — CHANGES_REQUESTED** [review]

**Review by jeremiahC — APPROVED** [review]

**Pull request merged**